src/main/java/com/aws/greengrass/mqttbridge/MQTTBridge.java

```diff
@@ -118,29 +120,37 @@ public void install() {
 
     @Override
     public void startup() {
+        String serverUri = Coerce.toString(config.findOrDefault(MQTTClient.DEFAULT_BROKER_URI,
+                KernelConfigResolver.CONFIGURATION_CONFIG_KEY, MQTTClient.BROKER_URI_KEY));
+        boolean ssl = serverUri.startsWith("ssl");
+
         try {
-            mqttClientKeyStore.init();
+            if (ssl) {
+                mqttClientKeyStore.init();
+            }
         } catch (CsrProcessingException | KeyStoreException | CsrGeneratingException e) {
             serviceErrored(e);
             return;
         }
 
         try {
-            kernel.locate(ClientDevicesAuthService.CLIENT_DEVICES_AUTH_SERVICE_NAME).getConfig()
-                    .lookup(RUNTIME_STORE_NAMESPACE_TOPIC, ClientDevicesAuthService.CERTIFICATES_KEY,
-                            ClientDevicesAuthService.AUTHORITIES_TOPIC).subscribe((why, newv) -> {
-                try {
-                    List<String> caPemList = (List<String>) newv.toPOJO();
-                    if (Utils.isEmpty(caPemList)) {
-                        logger.debug("CA list null or empty");
-                        return;
-                    }
-                    mqttClientKeyStore.updateCA(caPemList);
-                } catch (IOException | CertificateException | KeyStoreException e) {
-                    logger.atError("Invalid CA list").kv("CAList", Coerce.toString(newv)).log();
-                    serviceErrored(String.format("Invalid CA list. %s", e.getMessage()));
-                }
-            });
+            if (ssl) {
+                kernel.locate(ClientDevicesAuthService.CLIENT_DEVICES_AUTH_SERVICE_NAME).getConfig()
+                        .lookup(RUNTIME_STORE_NAMESPACE_TOPIC, ClientDevicesAuthService.CERTIFICATES_KEY,
+                                ClientDevicesAuthService.AUTHORITIES_TOPIC).subscribe((why, newv) -> {
+                            try {
```

Comment:
I believe the on change callback here will break things if the bridge is configured with ssl, then is changed to be without ssl.

Reply:
will take a closer look 👍